### PR TITLE
One-Time Checkout : PayPal Checkout

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
## What are you doing in this PR?

- Bring up the PayPal Checkout
- Allow payments

[**Trello Card**](https://trello.com](https://trello.com/c/WSpTNzCA/1086-generic-checkout-one-time-paypal))

## Screenshots
|before|after|
|-----|-----|
|image|image
|

### CODE REVIEW

File: support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx [hunk 1]
```diff
@@ -356,6 +358,35 @@ function OneTimeCheckoutComponent({
 			});
 		}
 
+		if (paymentMethod === 'PayPal') {
+			const paymentResult = await postOneOffPayPalCreatePaymentRequest({
+				currency: currencyKey,
+				amount: finalAmount ?? 0,
```
Comment (andrewHEguardian) [inline support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx:364]: We could probably handle this further up, if we don't have a `finalAmount` we shouldn't even start any of this payment stuff. And we could remove the if statement on line 414

**Review by andrewHEguardian — APPROVED** [review]
Great!

File: support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx [hunk 2]
```diff
@@ -356,6 +358,35 @@ function OneTimeCheckoutComponent({
...
+			// the PayPal flow on their site before we can actually try and execute the payment.
+			// So we drop a cookie which will be used by the /paypal/rest/return endpoint
+			// that the user returns to from PayPal, if payment is successful.
+			cookie.set(
+				'acquisition_data',
+				encodeURIComponent(JSON.stringify(acquisitionData)),
+			);
+
+			if (paymentResult.type === 'success') {
+				// setContributionAmount(finalAmount?.toString() ?? '');
```
Comment (GHaberis) [inline support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx:383]: Do we need this commented out line?

File: support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx (hunk 1 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx]: Yes thats a good idea, have moved the `finalAmount` check to top of `FormOnSubmit`. removed the lower check and failback shown above. Thanks

**Pull request merged**